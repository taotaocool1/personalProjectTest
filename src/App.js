import './App.css';
import React, { useState, useEffect } from 'react';
import { Table } from 'antd';
import TestRedux from './component/CountButton';
import './api/index.js';
import axios from 'axios';
function App() {
  const [state, setState] = useState(0);
  useEffect(() => {
    axios.get('/data', { dataType: 'json' }).then((res) => {
      console.log(res.data.list);
    });
  }, []);
  const dataSource = [
    {
      key: '1',
      name: '胡彦斌',
      age: 32,
      address: '西湖区湖底公园1号',
      a: '西湖区湖底公园1号',
      b: '西湖区湖底公园1号',
      c: '西湖区湖底公园1号',
      d: '西湖区湖底公园1号',
      e: '西湖区湖底公园1号',
      f: '西湖区湖底公园1号',
      g: '西湖区湖底公园1号',
      h: '西湖区湖底公园1号',
      i: '西湖区湖底公园1111111111111111111111号'
    },
    {
      key: '2',
      name: '胡彦祖',
      age: 42,
      address: '西湖区湖底公园1号',
    },
    {
      key: '3',
      name: '胡22',
      age: 42,
      address: '西湖区湖底公园1号',
    }
  ];
  
  const columns = [
    {
      title: '姓名',
      dataIndex: 'name',
      width: 40,
      key: 'name',
    },
    {
      title: '年龄',
      dataIndex: 'age',
      width: 40,
      key: 'age',
    },
    {
      title: '住址',
      dataIndex: 'a',
      width: 40,
      key: 'a',
    },
    {
      title: '住址',
      dataIndex: 'b',
      width: 40,
      key: 'b',
    }
    ,
    {
      title: '住址',
      dataIndex: 'c',
      width: 40,
      key: 'c',
    }
    ,
    {
      title: '住址',
      dataIndex: 'd',
      width: 40,
      key: 'd',
    }
    ,
    {
      title: '住址',
      dataIndex: 'e',
      width: 40,
      key: 'e',
    }
    ,
    {
      title: '住址',
      dataIndex: 'f',
      width: 20,
      key: 'f',
    }
    ,
    {
      title: '住址',
      dataIndex: 'g',
      width: 400,
      key: 'g',
    }
    ,
    {
      title: '住址',
      dataIndex: 'h',
      width: 300,
      key: 'h',
    }
    ,
    {
      title: '住',
      dataIndex: 'i',
      width: 100,
      key: 'i',
    },
    {}
  ];
  const setStateAsync = () => {
    setTimeout(() => {
        setState((state)=>++state);
    }, 1000)
  }
  return (
    <div className="App">
      {/* <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header> */}
      <a>123</a>
      <a>233</a>
      <a data-testid="display">{state}</a>
      <div style={{ width: 1200 }}><Table rowKey={(row) => row.key} dataSource={dataSource} columns={columns} /></div>
      <a>
          Learn React
      </a>
      <h3 data-testid="mainHeader">
          React Example
      </h3>
      <button data-testid="button1" onClick={()=>setState((state)=>++state)}>++</button>
      <button data-testid="button2" onClick={()=>setStateAsync()}>2</button>
      <div>
        <h1>这个是redux操作按钮数据</h1>
        <TestRedux />
      </div>
    </div>
  );
}
export default App;
// export default TestRedux;
