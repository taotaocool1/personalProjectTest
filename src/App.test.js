import { render, screen, fireEvent, cleanup, waitFor } from '@testing-library/react';
import App from './App';
import { Button } from 'antd';
afterEach(cleanup)
describe('App', () => {
  test('renders learn react link', async() => {
    const { getByTestId, getByText, queryByText, asFragment } = render(<App />); 
    // 生成快照验证，如果内容有修改需要输入u重新生成
    expect(asFragment()).toMatchSnapshot();
    const linkElement = getByText(/姓名/i);
    // 从页面全部doc获取有没有
    expect(linkElement).toBeInTheDocument();
    const el = queryByText('Hello');
    expect(el).toBeNull();
    expect(getByTestId("mainHeader")).toHaveTextContent("React Example");
    expect(getByTestId('display')).toHaveTextContent('0');
    // 模拟点击该按钮
    fireEvent.click(getByTestId('button1'));
    expect(getByTestId('display')).toHaveTextContent('1');

    // 查看点击按钮，异步获取是否正确
    fireEvent.click(getByTestId('button2'));
    // await 里面获取的是页面异步操作的数值应该返回的值，比如之前数值1，异步操作后应该是2，这里填写2表示我预知需要的结果
    const counter = await waitFor(() => getByText('2')) 
    // 判断预知结果是否正确
    expect(counter).toHaveTextContent('2');
    expect(getByTestId('button2')).not.toHaveAttribute('disabled');

  });
});
