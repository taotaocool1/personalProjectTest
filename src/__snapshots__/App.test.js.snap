// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders learn react link 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <a>
      123
    </a>
    <a
      data-testid="display"
    >
      0
    </a>
    <div
      style="width: 1200px;"
    >
      <div
        class="ant-table-wrapper"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup>
                      <col
                        style="width: 40px; min-width: 40px;"
                      />
                      <col
                        style="width: 40px; min-width: 40px;"
                      />
                      <col
                        style="width: 40px; min-width: 40px;"
                      />
                      <col
                        style="width: 40px; min-width: 40px;"
                      />
                      <col
                        style="width: 40px; min-width: 40px;"
                      />
                      <col
                        style="width: 40px; min-width: 40px;"
                      />
                      <col
                        style="width: 40px; min-width: 40px;"
                      />
                      <col
                        style="width: 20px; min-width: 20px;"
                      />
                      <col
                        style="width: 400px; min-width: 400px;"
                      />
                      <col
                        style="width: 300px; min-width: 300px;"
                      />
                      <col
                        style="width: 100px; min-width: 100px;"
                      />
                    </colgroup>
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                        >
                          姓名
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          年龄
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          住址
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          住址
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          住址
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          住址
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          住址
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          住址
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          住址
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          住址
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          住
                        </th>
                        <th
                          class="ant-table-cell"
                        />
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="1"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          胡彦斌
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          32
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          西湖区湖底公园1号
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          西湖区湖底公园1号
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          西湖区湖底公园1号
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          西湖区湖底公园1号
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          西湖区湖底公园1号
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          西湖区湖底公园1号
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          西湖区湖底公园1号
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          西湖区湖底公园1号
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          西湖区湖底公园1111111111111111111111号
                        </td>
                        <td
                          class="ant-table-cell"
                        />
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="2"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          胡彦祖
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          42
                        </td>
                        <td
                          class="ant-table-cell"
                        />
                        <td
                          class="ant-table-cell"
                        />
                        <td
                          class="ant-table-cell"
                        />
                        <td
                          class="ant-table-cell"
                        />
                        <td
                          class="ant-table-cell"
                        />
                        <td
                          class="ant-table-cell"
                        />
                        <td
                          class="ant-table-cell"
                        />
                        <td
                          class="ant-table-cell"
                        />
                        <td
                          class="ant-table-cell"
                        />
                        <td
                          class="ant-table-cell"
                        />
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
            <ul
              class="ant-pagination ant-table-pagination"
              unselectable="unselectable"
            >
              <li
                aria-disabled="true"
                class="ant-pagination-prev ant-pagination-disabled"
                title="Previous Page"
              >
                <a
                  class="ant-pagination-item-link"
                  disabled=""
                >
                  <span
                    aria-label="left"
                    class="anticon anticon-left"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="left"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                      />
                    </svg>
                  </span>
                </a>
              </li>
              <li
                class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                tabindex="0"
                title="1"
              >
                <a>
                  1
                </a>
              </li>
              <li
                aria-disabled="true"
                class="ant-pagination-next ant-pagination-disabled"
                title="Next Page"
              >
                <a
                  class="ant-pagination-item-link"
                  disabled=""
                >
                  <span
                    aria-label="right"
                    class="anticon anticon-right"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="right"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                      />
                    </svg>
                  </span>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <a>
      Learn React
    </a>
    <h3
      data-testid="mainHeader"
    >
      React Example
    </h3>
    <button
      data-testid="button1"
    >
      ++
    </button>
    <button
      data-testid="button2"
    >
      2
    </button>
  </div>
</DocumentFragment>
`;
